frontend/lib/ui/dashboard/artefact_dialog.dart

```diff
@@ -274,6 +276,23 @@ class _TestExecutionView extends StatelessWidget {
     final jenkinsLink = testExecution.jenkinsLink;
     final c3Link = testExecution.c3Link;
 
+    // Split the test result in separate categories based on Status (PASS/FAIL/SKIP)
+    final Map<TestResultStatus, List<TestResult>> testResultGroups = {};
```

Comment:
you could use [groupBy](https://api.flutter.dev/flutter/package-collection_collection/groupBy.html)